packages/outputarea/src/model.ts

```diff
@@ -309,44 +325,43 @@ export class OutputAreaModel implements IOutputAreaModel {
     // Consolidate outputs if they are stream outputs of the same kind.
     if (
       nbformat.isStream(value) &&
-      this._lastStream &&
-      value.name === this._lastName &&
+      value.name === this._lastStreamName &&
       this.shouldCombine({
         value,
         lastModel: this.list.get(this.length - 1)
       })
     ) {
-      // In order to get a list change event, we add the previous
-      // text to the current item and replace the previous item.
-      // This also replaces the metadata of the last item.
-      this._lastStream += value.text as string;
-      this._lastStream = Private.removeOverwrittenChars(this._lastStream);
-      value.text = this._lastStream;
-      const item = this._createItem({ value, trusted });
-      const index = this.length - 1;
-      const prev = this.list.get(index);
-      this.list.set(index, item);
-      prev.dispose();
+      // We append the new text to the current text.
+      // This creates a text change event.
+      const prev = this.list.get(this.length - 1) as IOutputModel;
+      const curText = prev.observableText!;
+      const newText = value.text as string;
+      Private.addText(curText, newText);
       return this.length;
     }
 
+    let newText = '';
     if (nbformat.isStream(value)) {
-      value.text = Private.removeOverwrittenChars(value.text as string);
+      newText = value.text as string;
```

Comment:
> I am fine with keeping it as-is and tackling later.

This is a pre-existing issue, but it looks like there should be no cast to `string` here; `isStream` is a type guard and correctly identifies that `value.text` exists and is of `string | string[]` type as per:

https://github.com/jupyterlab/jupyterlab/blob/6f202ff2d0ddf42414b09a27af1bbcb7954b51e7/packages/nbformat/src/index.ts#L458-L473

https://github.com/jupyterlab/jupyterlab/blob/6f202ff2d0ddf42414b09a27af1bbcb7954b51e7/packages/nbformat/src/index.ts#L65

```suggestion
      newText = Array.isArray(value.text) ? value.text.join('\n') : value.text;
```

Of note, `nbformat` source of truth schema also says that stream can be an array of strings https://github.com/jupyter/nbformat/blob/ba2c6f53afbeb18f6fad0c9b9f67def34e757cb4/nbformat/v4/nbformat.v4.schema.json#L360-L363